### PR TITLE
added 1 bit Format Mono LSB support

### PR DESCRIPTION
**Pull request by ValkA**
I think I could use QImage::Format_Mono flag, but it wasn't working for me, so I wrote a code that converts it to Grayscale8.

### CODE REVIEW

File: librgb/pixel_formats.py [hunk 1]
```diff
@@ -19,6 +20,7 @@ class PixelFormats(object):
     RGBA8888x = 232
 
     _NAME_MAP = [
+        (GRAY1, '1:Image1Bit (1)', 'G1'),
```
Comment (rr-) [inline librgb/pixel_formats.py]:
```suggestion
        (GRAY1, '1:Binary (1)', 'G1'),
```

File: librgb/renderer.py [hunk 2]
```diff
@@ -50,10 +51,21 @@ def get_pixmap(self):
 
         stride = params.width * channels
         data_size = params.height * stride
+        if params.format == PixelFormats.GRAY1:
+            data_size = data_size / 8
```
Comment (rr-) [inline librgb/renderer.py]:
```suggestion
            data_size /= 8
```

File: librgb/renderer.py [hunk 3]
```diff
@@ -50,10 +51,21 @@ def get_pixmap(self):
...
         data = reader.get_padded_bytes(data_size)
         assert data is not None
         assert len(data) == data_size
 
+        if params.format == PixelFormats.GRAY1:
+            old_data = data
+            data = ''
+            data_size = data_size * 8
+            while len(old_data) > 0:
+                data = data + ''.join(''.join(chr(((ord(byte)&(1<<bit))>>bit)*255) for byte in old_data[:stride]) for bit in range(8))
```
Comment (rr-) [inline librgb/renderer.py]:
```suggestion
                data += ''.join(
                    ''.join(
                        chr(((ord(byte) & (1 << bit)) >> bit) * 255)
                        for byte in old_data[:stride]
                    )
                    for bit in range(8)
                )
```

**Review by rr- — CHANGES_REQUESTED** [review]
Cosmetic changes only.
It would be great to offer faster alternative if numpy is available similar to how brightness adjustment looks like, but it's not a requirement.

I'll take a look if I can get it to work with Qt's mono pixel format later, for now we'll merge it like this (after the formatting changes).

**Comment by ValkA** [comment]
Have you tested it? I think python doesn't support += and /=. Anyway, if it works I don't care

**Comment by rr-** [comment]
>Have you tested it? I think python doesn't support += and /=.

…
Yes, it does.
https://docs.python.org/3/reference/grammar.html `augassign`

**Comment by rr-** [comment]
It doesn't work for me…

1. Data size not divisible by 8 lead to a crash
2. 
```
  File "/home/rr-/src/maintained/ida-images/librgb/renderer.py", line 71, in <genexpr>
    for bit in range(8)
  File "/home/rr-/src/maintained/ida-images/librgb/renderer.py", line 69, in <genexpr>
    for byte in old_data[:stride]
TypeError: ord() expected string of length 1, but int found
```

I'll try to fix the issue within this PR.

**Comment by rr-** [comment]
LMK if this works for you.

**Comment by rr-** [comment]
I'm assuming that's a "yes".
(It's working on my end.)

**Pull request merged**